Source/ServerTrustEvaluation.swift

```diff
@@ -65,7 +81,8 @@ public protocol ServerTrustEvaluating {
     ///   - trust: The `SecTrust` value to evaluate.
     ///   - host:  The host for which to evaluate the `SecTrust` value.
     /// - Returns: A `Bool` indicating whether the evaluator considers the `SecTrust` value valid for `host`.
-    func evaluate(_ trust: SecTrust, forHost host: String) -> Bool
+    /// - Throws:  An `AFError.serverTrustEvaluationFailed` with an associated `ServerTrustFailureReason`.
+    func evaluate(_ trust: SecTrust, forHost host: String) throws -> Bool
     #endif
 }
```

Comment:
You could create a second protocol for Evaluators to handle `performDefaultEvaluation` and `validateHost` logic. Looks duplicated in `DefaultTrustEvaluator`, `RevocationTrustEvaluator`, `PinnedCertificatesTrustEvaluator`, and `PublicKeysTrustEvaluator`. Could be something like:

```swift
public protocol CommonServerTrustEvaluating: ServerTrustEvaluating {
    #if os(Linux)
    // Implement this once Linux has API for evaluating server trusts.
    #else
    var performDefaultValidation: Bool { get }
    var validateHost: Bool { get }

    func performDefaultValidationIfEnabled() throws -> Bool
    func validateHostIfEnabled() throws -> Bool
    #endif
}
```

Then in all the evaluations, this:

```swift
if performDefaultValidation {
    try trust.validate(policy: .default) { (status, result) in
        AFError.serverTrustEvaluationFailed(reason: .defaultEvaluationFailed(output: .init(host, trust, status, result)))
    }
}
        
if validateHost {
    try trust.validate(policy: .hostname(host)) { (status, result) in
        AFError.serverTrustEvaluationFailed(reason: .hostValidationFailed(output: .init(host, trust, status, result)))
    }
}
```

Could become this:

```swift
try performDefaultValidationIfEnabled()
try validateHostIfEnabled()
```

Or something also those lines as far as naming. Not critical or anything, just spotted a way to DRY it up.

Reply:
I've DRY'd up a little bit more by moving the actual evaluations into `SecTrust` extensions, but I don't think complicating the protocols to get rid of the `if` checks would be worth it. Take another look.